httpreq/httpreq.go

```diff
@@ -76,6 +90,18 @@ func (r *HttpReq) WithToken(token string) *HttpReq {
 	return r
 }
 
+func (r *HttpReq) WithTransport(transport *http.Transport) *HttpReq {
```

Comment:
Do not change `defaultClient`.
I think it is better `CallWithTransport` or `CallWithHttpClient` rather then `WithTransport`.

Reply:
ok,I think so